### DISCUSSION TITLE
Explicitly pass session / auth data (as opposed to passing it through the Context)

### DISCUSSION THREAD

**Discussion by josephbuchma** [ogen-go/ogen#1122] [Q&A]
Is there a way to add additional properties to handler parameters that will be populated in the middleware? As far as I can tell, right now, the only way to have session data available in the handler is to add it to the context in the middleware.

**Comment by ghost** [comment]
For now, the only way is to pass it through context.

**Reply by josephbuchma** [reply]
Are you aware of any other OpenAPI code generator that has this feature? What would it take to implement it in ogen?

**Reply by ghost** [reply]
[`go-swagger` has concept of principal](https://goswagger.io/tutorial/authentication/#authentication-sample).

But it is OpenAPI 2.0 (Swagger) generator, it does not support 3.0.

I don't know any OpenAPI 3 code generator for Go that has feature like this.
